### ISSUE TITLE
XrdCl::zip file extraction segmentation fault

### ISSUE THREAD

**Issue by adriansev** [xrootd/xrootd#1393]
@simonmichal Hi! i just received a bug report with i believe use xrootd 4.12.5
which can be reproduced with example_cmd and meta file from https://cernbox.cern.ch/index.php/s/4lhS4eE322CuNRx
also there are the core files for both python usage and xrdcp usage.

**Comment by adriansev** [comment]
but i reproduced the crash also with 5.0.3

**Comment by simonmichal** [comment]
@adriansev : thanks for reporting the problem! I'll have a look!

**Comment by adriansev** [comment]
@simonmichal so, we did a bit of scanning of xrootd versions and it seems that all versions do the same thing when accessing some of these files (but not all !!!) on this particular EOS server (we are looking to find other cases to have more overall perspective). Still, no matter the problem i would say that xrootd client still have to treat it graciously and not segfault so i will keep this issue open.

**Comment by simonmichal** [comment]
@adriansev : hmm I cannot reproduce the problem, doing

`xrdcp -f ./fc46df15-6c71-5ac0-9003-1c40ed33e28f.meta4  .`

works perfectly (I tried with both 5.0.3 and 4.12.5), it is the `fc46df15-6c71-5ac0-9003-1c40ed33e28f.meta4` file, isn't it?

Could you give me more details on how you are running your test? Could you try running `xrdcp` in `valgrind`?

**Comment by adriansev** [comment]
@simonmichal the actual problem is with extraction of the file from the zip. take a look at example_cmd file from the cernbox share.

**Comment by simonmichal** [comment]
OK, all clear now!

**Comment by adriansev** [comment]
another thing that i just realized : all other files from archive have no problems, only the rec.log goes to seg fault.
the archive seem ok, i downloaded and `zip -T` was ok

**Comment by adriansev** [comment]
ok, found another run, on xrootd plain server (not eos) with the same symptoms .. also, for the record the content of archive looks like this (the archive of the rec.log from cernbox share)
```
unzip -l log_archive
Archive:  log_archive
Length      Date    Time    Name
---------  ---------- -----   ----
335713  08-28-2015 22:10   aod.log
2991  08-28-2015 21:48   fileinfo.log
15901  08-28-2015 21:54   mergeQA.log
2390688  08-28-2015 21:38   qa_grp0.log
244506  08-28-2015 21:42   qa_grp1.log
830595  08-28-2015 21:44   qa_grp2.log
451438  08-28-2015 21:47   qa_grp3.log
250468  08-28-2015 21:49   qa_grp4.log
36982932  08-28-2015 21:11   rec.log
1045  08-28-2015 22:10   stderr.log
13523  08-28-2015 22:12   stdout.log
16285  08-28-2015 22:07   syswatch.log
13834  08-28-2015 21:25   tag.log
---------                     -------
41549919                     13 files
```

**Comment by simonmichal** [comment]
@adriansev : the problem should be fixed in the following 3 commits: 928f0d173c33e17faad2bb353d9ff3ee1b3c182f, 09ac1c652b1d1a38890793e1c8fb5f1a48f22520, d83bffcf7e76dee297a5635cf85815280ef9b2fe

I tested it with several ZIP files, but to be sure, could you give it a try before we close it?

**Comment by adriansev** [comment]
@simonmichal i can confirm that it is working. the same file that used to generate segfault no longer exhibit the behavior.   
Thanks a lot!!
